MailCore/Cache/MailboxManager.swift

```diff
@@ -694,28 +694,39 @@ public class MailboxManager: ObservableObject {
 
     private func createMultiMessagesThreads(messageByUids: MessageByUidsResult, folder: Folder, using realm: Realm) {
         var threadsToUpdate = Set<Thread>()
-        try? realm.safeWrite {
-            for message in messageByUids.messages {
-                if realm.object(ofType: Message.self, forPrimaryKey: message.uid) == nil {
-                    message.inTrash = folder.role == .trash
-                    message.computeReference()
-                    let existingThreads = Array(realm.objects(Thread.self)
-                        .where { $0.messageIds.containsAny(in: message.linkedUids) })
-
-                    if let newThread = createNewThreadIfRequired(for: message, folder: folder, existingThreads: existingThreads) {
-                        threadsToUpdate.insert(newThread)
-                    }
+        do {
+            let regex = try NSRegularExpression(pattern: Constants.referenceRegex)
```

Comment:
Place the whole NSRegularExpression in a static constant to avoid recreating it for each thread